Sources/ApolloCodegenLib/Templates/MockObjectTemplate.swift

```diff
@@ -56,21 +59,38 @@ struct MockObjectTemplate: TemplateRenderer {
       
       \(accessControlModifier(for: .parent))\
       extension Mock where O == \(objectName) {
-        \(conflictingFieldNameProperties(fields))
+        \(conflictingFieldNameProperties(fields.map { $0.0 }))
         convenience init(
           \(fields.map { """
-            \($0.propertyName)\(ifLet: $0.initializerParameterName, {" \($0)"}): \($0.mockType)? = nil
+            \($0.0.propertyName)\(ifLet: $0.0.initializerParameterName, {" \($0)"}): \($0.0.mockType)? = nil
             """ }, separator: ",\n")
         ) {
           self.init()
-          \(fields.map { "_set(\($0.initializerParameterName ?? $0.propertyName), for: \\.\($0.propertyName))" }, separator: "\n")
+          \(fields.map {
+            return "_set\(mockFunctionDescriptor($0.1))(\($0.0.initializerParameterName ?? $0.0.propertyName), for: \\.\($0.0.propertyName))"
+          }, separator: "\n")
         }
       }
       """) : TemplateString(stringLiteral: "")
     )
     
     """
   }
+  
+  private func mockFunctionDescriptor(_ graphQLType: GraphQLType) -> String {
+    switch graphQLType {
+      case .list(_):
+        return "List"
+      case .scalar(_), .enum(_):
+        return "Scalar"
+      case .entity(_):
+        return "Entity"
+      case .inputObject(_):
+        return ""
```

Comment:
This should probably be a `preconditionFailure` as if we return an empty string here, you're just going to get code that doesn't compile. Better to fail during generation with an error message that people can see and put in a bug report for us.